File: email_notifications/mailgun/nodejs/send_email.js
const sgMail      = require('mailgun-js');
const ironWorker  = require('iron_worker');

const mailgun     = require('mailgun-js')({apiKey: process.env.MAILGUN_API_KEY, domain: ironWorker.params().domain});

const data = {
    to:      ironWorker.params().to,
    from:    ironWorker.params().from,
    subject: ironWorker.params().subject,
    html:    ironWorker.params().html,
};

console.log(`--> Sending email obj: ${JSON.stringify(data)}`);

mailgun.messages().send(data, function (error, body) {
    console.log(error);
    if (error) {
        console.error(error);
        process.exit(1);
    } else {
        console.log('--> Email has been sent!');
        console.log(body);
    }
});

File: email_notifications/mailgun/ruby/send_email.rb
require_relative 'bundle/bundler/setup'
require 'iron_worker'
require 'mailgun'

# First, instantiate the Mailgun Client with your API key
mg_client = Mailgun::Client.new ENV['MAILGUN_API_KEY']

# Define your message parameters
message_params = {:from    => IronWorker.payload['from'],
                  :to      => IronWorker.payload['to'],
                  :subject => IronWorker.payload['subject'],
                  :text    => IronWorker.payload['body']}

# Send your message through the client
response = mg_client.send_message IronWorker.payload['domain'], message_params

puts response.body.to_s